File: 6/longest_continuous_increasing_subseq.go
package main

import "fmt"

func max(a, b int) int {
	if a > b {
		return a
	}
	return b

}

func findLengthOfLCIS(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	maxLen := 1
	currentLen := 1

	for i := 1; i < len(nums); i++ {
		if nums[i] > nums[i-1] {
			currentLen++
		} else {
			maxLen = max(maxLen, currentLen)
			currentLen = 1
		}
	}

	maxLen = max(maxLen, currentLen)
	return maxLen
}

func main() {
	res := findLengthOfLCIS([]int{3, 4, 5, 2, 3})
	fmt.Println(res)
}

File: 6/longest_continuous_increasing_subseq.py
class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        if not nums:
            return 0
        max_len = 0
        current_len = 1
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                current_len += 1
            else:
                max_len = max(max_len, current_len)
                current_len = 1
        max_len = max(max_len, current_len)
        return max_len
                
